simpleble/src/backends/linux/AdapterLinux.cpp

```diff
@@ -1,46 +1,29 @@
+#include <thread>
+
+#include <simpleble/Peripheral.h>
+
 #include "AdapterLinux.h"
-#include "BackendBluez.h"
 #include "BuildVec.h"
 #include "BuilderBase.h"
 #include "CommonUtils.h"
 #include "PeripheralLinux.h"
 
 using namespace SimpleBLE;
 
-std::vector<std::shared_ptr<AdapterBase>> AdapterBase::get_adapters() {
-    std::vector<std::shared_ptr<AdapterBase>> adapter_list;
-    auto internal_adapters = Bluez::get()->bluez.get_adapters();
-    for (auto& adapter : internal_adapters) {
-        adapter_list.push_back(std::make_shared<AdapterBase>(adapter));
-    }
-    return adapter_list;
-}
-
-bool AdapterBase::bluetooth_enabled() {
-    bool enabled = false;
-
-    auto internal_adapters = Bluez::get()->bluez.get_adapters();
-    for (auto& adapter : internal_adapters) {
-        if (adapter->powered()) {
-            enabled = true;
-            break;
-        }
-    }
-
-    return enabled;
-}
+bool AdapterLinux::bluetooth_enabled() { return adapter_->powered(); }
```

Comment:
This doesn't match what the original implementation did. Unfortunately we need to check every individual adapter.

Reply:
The thing with bluetooth_enabled is that there are different levels of granularity supported by different systems. Up to now, it was a static method, which means it reported the system-wide (and thus backend-wide, because only one backend could be enabled) setting. 

In this PR I split the adapter-specific and backend-general enabled settings (which for some systems is the same).

On linux, it seems like one can query the powered status of individual adapters separately and the implementation was emulating a system-wide setting by returning true if at least one adapter was enabled. A problem with this is that the specific adapter corresponding to the Adapter object that the user has may not be the one that is enabled.

Reply:
Yeah, we'll need to redesign this at some point, but I think for now your approach is fine, let's ship it.